### PR TITLE
chore(ci): osx ARM64 build by removing envs

### PR DESCRIPTION
**Pull request by leet4tari**
Description
Remove build envs used for croaring with cross-compiling that was breaking OSX ARM64 builds.
Re-enabled OSX ARM64 builds with fix in place.
Fix some missing tari to minotari rename operations, also helps with OSX PKG assembly.

Motivation and Context
Bring back the OSX ARM64 builds and also fix all OSX PKG assembly.

How Has This Been Tested?
Tested build in local fork and pkg install locally.

What process can a PR reviewer use to test or verify this change?
Note that OSX ARM64 builds successful and pkg assembly without errors.

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

File: .github/workflows/base_node_binaries.yml [hunk 1]
```diff
@@ -192,13 +188,16 @@ jobs:
...
 
       - name: Build rust binaries
-        env:
-          CARGO_UNSTABLE_SPARSE_REGISTRY: true
-          RUSTFLAGS: "-C target_cpu=${{ matrix.builds.target_cpu }}"
-          ROARING_ARCH: "${{ matrix.builds.target_cpu }}"
+        # env:
+        #   CARGO_UNSTABLE_SPARSE_REGISTRY: true
+        #   RUSTFLAGS: "-C target_cpu=${{ matrix.builds.target_cpu }}"
+        #   ROARING_ARCH: "${{ matrix.builds.target_cpu }}"
```
Comment (brianp) [inline .github/workflows/base_node_binaries.yml]: We need to leave commented out code?

**Review by brianp — CHANGES_REQUESTED** [review]
Mostly looks good. Lets try not to leave stuff that's commented out unless it's got an explicit reason to stay.

File: .github/workflows/base_node_binaries.yml (hunk 1 above)
Comment (leet4tari) [reply .github/workflows/base_node_binaries.yml]: Until all the builds and binaries have been fully tested, I would prefer too.

Comment (SWvheerden) [reply .github/workflows/base_node_binaries.yml]: I prefer we remove them. 
We have the git history if we want to retrieve them again. 
Commented out code tends to live forever in the code.

**Review by SWvheerden — COMMENTED** [review]
Looks good, I would just prefer we remove that commented code

**Review by SWvheerden — APPROVED** [review]

**Pull request merged**